dotcom-rendering/src/components/SecureSignupIframe.importable.tsx

```diff
@@ -317,6 +318,42 @@ export const SecureSignupIframe = ({
 		resetIframeHeight();
 	};
 
+	const autofillEmailIfSignedIn = () => {
+		const { current: iframe } = iframeRef;
+		const emailFormField = iframe?.contentDocument?.querySelector(
+			'input[name="email"]',
+		);
+		const signUpButton = iframe?.contentDocument?.querySelector(
+			'button[type="submit"]',
+		);
+		const labelNodeList =
+			iframe?.contentDocument?.querySelectorAll('label');
+		const { idApiUrl } = window.guardian.config.page;
+
+		const fetchEmail = idApiUrl
+			? lazyFetchEmailWithTimeout(idApiUrl)
+			: undefined;
```

Comment:
None of this work is necessary if there’s either:
- no `idApiUrl` defined
- the email is not resolved

Is it worth converting this function to an asynchronous one and returning early if there are missing elements? For example:

```ts
const autofillEmailIfSignedIn = async () => {
	const fetchEmail = idApiUrl
		? lazyFetchEmailWithTimeout(idApiUrl)
		: undefined;
	if(!fetchEmail) return;
	const email = await fetchEmail()
	if(!email) return;
	const { current: iframe } = iframeRef;
	if(!iframe) return;
	/* etc. */
}
```

This can greatly reduce the indendation of your function.